Biz/Model/RealmOwner.pm

```diff
@@ -347,6 +353,40 @@ sub validate_login {
     return 'NOT_FOUND';
 }
 
+sub validate_password {
+    my($self, $clear_text) = @_;
+    return 'WEAK_PASSWORD'
+        if $clear_text eq $self->get('name')
+        || $clear_text eq $self->get('realm_id')
+        || _canonicalize($clear_text) eq _canonicalize($self->get('display_name'))
+        || _similar_to_email($self, $clear_text);
+    return;
+}
+
+sub _canonicalize {
+    my($value) = @_;
+    $value =~ s/\W+//g;
+    $value = lc($value);
+    return $value;
+}
+
+sub _similar_to_email {
+    my($self, $value) = @_;
```

Comment:
I think you need to downcase $value. Seems like you might want to canonicalize both.